js/application.js

```diff
@@ -16,27 +16,75 @@ const setDetailContainerHeight = (el, isOpen = false) => {
   el.style.setProperty("--height", (isOpen ? el.scrollHeight: 0) + "px");
 }
 
-document.getElementById("tutorial-button").addEventListener("click", function() {
-  setDetailContainerHeight(document.getElementById("tutorial"), true);
-  setDetailContainerHeight(document.getElementById("explanation"), false);
-  setDetailContainerHeight(document.getElementById("about"), false);
-});
+function sectionClick(id) {
+  let idBtn = id + "-button";
 
-document.getElementById("explanation-button").addEventListener("click", function() {
-  setDetailContainerHeight(document.getElementById("tutorial"), false);
-  setDetailContainerHeight(document.getElementById("explanation"), true);
-  setDetailContainerHeight(document.getElementById("about"), false);
-});
+  //LOGIC
+  // check if the element is already open ("active" class)
+  if (document.getElementById(idBtn).classList.contains("active")) {
+    // if it is, close it
+    setDetailContainerHeight(document.getElementById(id), false);
+    document.getElementById(idBtn).classList.remove("active");
+    return;
+  }
 
-document.getElementById("about-button").addEventListener("click", function() {
-  setDetailContainerHeight(document.getElementById("tutorial"), false);
-  setDetailContainerHeight(document.getElementById("explanation"), false);
-  setDetailContainerHeight(document.getElementById("about"), true);
-});
+  // STYLE
+  // remove other active classes
+  document.querySelectorAll(".expand-button").forEach(function(el) {
+    if (el.id == idBtn) {
+      el.classList.add("active");
+    }
+    else {
+      el.classList.remove("active");
+    }
+  });
 
+  // FUNCTION
+  // close all elements
+  document.querySelectorAll(".detail-container").forEach(function(el) {
+    // if el.id != id, close it
+    setDetailContainerHeight(el, el.id == id);
+  });
 
+  // scroll as element is opening
+  setTimeout(function() {
+    document.getElementById(id).scrollIntoView({
+      behavior: "smooth",
+      block: "start",
+      inline: "nearest"
```

Comment:
The `block:"start"` and `inline:"nearest"` properties can be removed as those are the default values for `scrollIntoView`

Reply:
done